src/v/raft/consensus.cc

```diff
@@ -293,8 +293,15 @@ consensus::success_reply consensus::update_follower_index(
 
 void consensus::maybe_promote_to_voter(model::node_id id) {
     (void)ss::with_gate(_bg, [this, id] {
+        const auto& latest_cfg = _configuration_manager.get_latest();
+
+        // node is no longer part of current configuration, skip promotion
+        if (!latest_cfg.current_config().contains(id)) {
+            return ss::now();
+        }
+
         // is voter already
-        if (config().is_voter(id)) {
+        if (latest_cfg.is_voter(id)) {
```

Comment:
if it is possible that `maybe_promote_to_voter` is called for a node id that is already a voter, is it possible that it is called for a node id that is already a voter _and is also_ not in the current configuration? would that need a special case like demotion to candidate?

Reply:
What do you mean by demotion to `candidate` ?

Reply:
looking back at this i don't think i have the same question now. i must have be confused about the combination of conditions.